attribute_json/models/attribute_attribute.py

```diff
@@ -0,0 +1,114 @@
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import lazy_property
+from odoo.tools.translate import _
+
+
+class AttributeAttribute(models.Model):
+    _inherit = "attribute.attribute"
+    _description = "Attribute"
+
+    nature = fields.Selection(
+        selection_add=[("json_postgresql", "Json_postgresql")],
+        ondelete={"json_postgresql": "set default"},
+    )
+
+    attribute_type = fields.Selection(
+        selection_add=[("monetary", "Monetary"), ("selection", "Selection")],
+        ondelete={"monetary": "cascade"},
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        attributes = super().create(vals_list)
+        attributes._update_relation_model_attribute()
+        return attributes
+
+    def _update_relation_model_attribute(self):
+        """
+        Update the relation model with the new attribute
+        We follow the same logic as the odoo core to create the field
+        """
+        to_refresh = False
+        for attribute in self:
+            if attribute.nature == "json_postgresql":
+                to_refresh = True
+                Record = self.env[attribute.relation_model_id.model]
+                fields_data = self.env["ir.model.fields"]._get_manual_field_data(
+                    Record._name
+                )
+                field_data = fields_data.get(attribute.name)
+                try:
+                    field = self.env["ir.model.fields"]._instanciate(field_data)
+                    if field is not None:
+                        # Not the best way.
+                        # Odoo should add these attribute
+                        setattr(field, "depends", tuple())
+                        setattr(field, "depends_context", tuple())
```

Comment:
This should be done by Odoo. But when a json field field is created for the first time the depends and depends_context attribute aren't set on it. When Odoo restart the depends is set by Odoo. I tried to find the difference between the implementation and how Odoo loads field from DB with no success. 

Any help is welcome :)